airflow/models.py

```diff
@@ -1996,7 +1996,14 @@ def execute(self, context):
 
         Refer to get_template_context for more context.
         """
-        raise NotImplementedError()
+        if self.dag:
+            os.environ['AIRFLOW_DAG_ID'] = self.dag.dag_id
+        dagrun = context['dag_run']
```

Comment:
@aoen this is the correct way to get a dag run yes? As discussed it is needed in order to retrieve the execution date.